equistore-torch/src/labels.cpp

```diff
@@ -265,6 +265,73 @@ LabelsHolder LabelsHolder::to_owned() const {
     }
 }
 
+TorchLabels LabelsHolder::append(std::string name, torch::Tensor values) {
+    return this->insert(this->size(), std::move(name), std::move(values));
+}
+
+
+TorchLabels LabelsHolder::insert(int64_t index, std::string name, torch::Tensor values) {
+    auto names = this->names();
+
+    auto it = std::begin(names) + index;
+    names.insert(it, name);
+
+    if (values.sizes().size() != 1) {
+        C10_THROW_ERROR(
+            ValueError,
+            "`values` must be a 1D tensor"
+        );
+    }
+
+    auto old_values = this->values();
+    auto first = old_values.index({torch::indexing::Slice(), torch::indexing::Slice(0, index)});
+    auto second = old_values.index({torch::indexing::Slice(), torch::indexing::Slice(index)});
+
+    auto new_values = torch::hstack({first, values.reshape({1, static_cast<int64_t>(values.size(0))}), second});
```

Comment:
```suggestion
    auto new_values = torch::hstack({first, values.reshape({values.size(0), 1}), second});
```

`values.size(0)` should already be a 64-bit int. Also, I think the shape was the wrong way around?

Reply:
yes it is the other way around. But why is this also working...

Reply:
not sure, maybe some torch magic/dispatch?